### ISSUE TITLE
panos_object - Error: Parent device group 'Shared' is not recognised as valid device group

### ISSUE THREAD

**Issue by smatt241** [PaloAltoNetworks/ansible-pan#44]
failed: [10.x.x.x -> localhost] (item={u'name': u'test4', u'device_group': u'Shared'}) => {
    "changed": false,
    "failed": true,
    "invocation": {
        "module_args": {
            "address": null,
            "address_type": "ip-netmask",
            "addressgroup": null,
            "addressobject": null,
            "api_key": null,
            "color": null,
            "description": null,
            "destination_port": null,
            "devicegroup": "Shared",
            "dynamic_value": null,
            "ip_address": "10.x.x.x",
            "operation": "add",
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "protocol": null,
            "servicegroup": null,
            "serviceobject": null,
            "services": null,
            "source_port": null,
            "static_value": null,
            "tag_name": "test4",
            "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    },
    "item": {
        "device_group": "Shared",
        "name": "test4"
    },
    "msg": "'Shared' device group not found in Panorama. Is the name correct?"
}

**Comment by niemesrw** [comment]
Ah yes, I'm seeing this as well. Upvoting for a fix please.

**Comment by btorresgil** [comment]
Have you tried naming the devicegroup 'shared' (all lower case)?

**Comment by niemesrw** [comment]
Just attempted, here's the text of the error:

    "msg": "'shared' device group not found in Panorama. Is the name
correct?"


On Tue, Mar 13, 2018 at 10:06 AM, Brian Torres-Gil <notifications@github.com

**Comment by niemesrw** [comment]
FWIW, when I put a valid device group in the playbook it works just fine:

changed: [ussd-prd-pano01] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "address": "35.154.2.119",
            "address_type": "ip-netmask",
            "addressgroup": null,
            "addressobject": "netskope-proxy-35.154.2.119",
            "api_key": null,
            "color": null,
            "description": "NetSkope Proxy Address 35.154.2.119",
            "destination_port": null,
            "devicegroup": "San Diego HQ",
            "dynamic_value": null,
            "ip_address": "ussd-prd-pano01",
            "operation": "update",
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "protocol": null,
            "servicegroup": null,
            "serviceobject": null,
            "services": null,
            "source_port": null,
            "static_value": null,
            "tag_name": null,
            "username": "svc_ansible"
        }
    },
    "msg": "Object 'netskope-proxy-35.154.2.119' successfully updated."
}

**Comment by btorresgil** [comment]
Thanks for trying.  Taking a look at this.

**Comment by shinmog** [comment]
This should work properly after updating pandevice to 0.6.0 or later.

**Comment by dannypetrov** [comment]
Same error applies for panos_security_rule. I have a devicegroup called "Shared" which is shared among all firewalls. When I try to apply a shared rule I get exactly the same error :/

**Comment by mcanetto** [comment]
I haven't tried with the panos_security_rule, but with panos_object I got this to work by leaving the devicegroup off of the playbook entirely. If it isn't specified it seems Panorama assumes Shared.

**Comment by shinmog** [comment]
That's correct:  this is because with config items that go into a vsys, on a firewall the default is to put it in vsys1, but on Panorama the default is the `shared` scope.

**Comment by btorresgil** [comment]
Closed by a different repo, re-opening.

**Comment by skathiresan-hw** [comment]
when you config for a firewall, by default it goes to vsys1, is it possible to create a shared object in a mulitvsys environment?
